### DISCUSSION TITLE
Unable to obtain updated version

### DISCUSSION THREAD

**Discussion by airsoup** [fwupd/fwupd#5820] [Q&A] [answered]
Hi, There is a Linux host that can recognize EM05G devices using the `get-device` command, but the fwupdmgr `get-releases` command cannot obtain an updated version. Is there any missing configuration?
![IMG_20230515_114853](https://github.com/fwupd/fwupd/assets/133010145/eb459fd7-937e-490c-ad1a-46e09a32a087)

**Comment by hughsie** [comment]
All the Quectel updates are embargoed on the LVFS for some reason. I guess you should ask Quectel why.

**Comment by airsoup** [comment]
Do different types of firmware rely on the same library files, or do some firmware require specific libraries to identify or update devices? There is another issue. I would like to install an LVFS environment in a new virtual machine. Where can I find the complete software installation process?

**Reply by hughsie** [reply]
> Do different types of firmware rely on the same library files

No, although two of the plugins are optional -- modem_manager and flashrom -- but both are installed by default unless you manually remove them.

> Where can I find the complete software installation process

Not sure how to answer this; fwupd is installed by default on basically all Linux distros.

**Comment by airsoup** [comment]
If you use apt-get directly, the installed version will be relatively low and firmware cannot be upgraded.

I tried to follow the installation method on the network, which used Mason and Ninja, and also installed many library files, but the installation still failed. -_-|||

Are there any installation guidelines for systems without pre installed FWUPD?

**Reply by hughsie** [reply]
> Are there any installation guidelines for systems without pre installed FWUPD?

Does https://lvfs.readthedocs.io/en/latest/custom-plugin.html#building-fwupd help?

**Reply by airsoup** [reply, marked as answer]
After installing many library files and reinstalling fwupd multiple times, we are finally able to update the firmware. Thank you for your help and have a pleasant time!
